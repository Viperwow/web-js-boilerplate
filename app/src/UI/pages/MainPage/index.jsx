// Vendors
import React, {Component} from 'react';
// Constants
const MAIN_PAGE_PLACEHOLDER = 'MAIN_PAGE_PLACEHOLDER';

class MainPage extends Component {
  render() {
    return (
      <div className="page">
        {MAIN_PAGE_PLACEHOLDER}
      </div>
    );
  }
}

export default MainPage;
