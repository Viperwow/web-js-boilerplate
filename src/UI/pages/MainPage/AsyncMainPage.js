import loadable from 'src/common/loadable';
import Loader from 'src/UI/components/Loader';

const DynamicMainPage = loadable({
  importer: () => import('src/UI/pages/MainPage/MainPageContainer' /* webpackChunkName: "MainPage" */), // eslint-disable-line max-len
  LoaderComponent: Loader,
});

export default DynamicMainPage;
