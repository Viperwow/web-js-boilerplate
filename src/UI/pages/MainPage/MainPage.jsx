import React from 'react';
import {Link} from 'react-router-dom';

const MainPage = ({text}) => (
  <div className="page">
    {text}
    <ul>
      <li>
        <Link to="/2" replace>
          2
        </Link>
      </li>
      <li>
        <Link to="/3" replace>
          3
        </Link>
      </li>
      <li>
        <Link to="/not-found" replace>
          4
        </Link>
      </li>
    </ul>
  </div>
);

export default MainPage;
