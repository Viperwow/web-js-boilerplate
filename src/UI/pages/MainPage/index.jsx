// Vendors
import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class MainPage extends Component {
  render() {
    const {text} = this.props;

    return (
      <div className="page">
        {text}
        <ul>
          <li>
            <Link to="/2" replace>
              2
            </Link>
          </li>
          <li>
            <Link to="/3" replace>
              3
            </Link>
          </li>
          <li>
            <Link to="/4" replace>
              4
            </Link>
          </li>
        </ul>
      </div>
    );
  }
}

export default MainPage;
