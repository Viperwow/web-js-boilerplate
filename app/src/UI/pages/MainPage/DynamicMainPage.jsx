import React from 'react';
import dynamic from 'src/common/dynamic';

const DynamicMainPage = () => {
  const Component = dynamic(
    import('src/UI/pages/MainPage' /* webpackChunkName: "MainPage" */),
  );

  return <Component />;
};

export default DynamicMainPage;
