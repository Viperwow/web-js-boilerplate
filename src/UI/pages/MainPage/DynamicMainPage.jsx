// HoC
import loadable from 'src/common/loadable';

const DynamicMainPage = loadable(
  import('src/UI/pages/MainPage' /* webpackChunkName: "MainPage" */),
)();

export default DynamicMainPage;
