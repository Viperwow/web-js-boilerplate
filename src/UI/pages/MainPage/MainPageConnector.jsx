import connectQuery from 'src/common/connectQuery';
import withOnError from 'src/common/withOnError';
import withTranslate from 'src/common/withTranslate';
import {compose} from 'src/helpers/utility';
import {LOCALE_QUERY} from 'src/queries';
import MainPage from './MainPage';

const mapStateToProps = (_, {t}) => ({
  text: t('pages.main'),
});

export default compose(
  withTranslate,
  connectQuery(LOCALE_QUERY, mapStateToProps),
  withOnError,
)(MainPage);
