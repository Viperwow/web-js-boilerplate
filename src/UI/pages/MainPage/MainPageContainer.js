import connectQuery from 'src/common/connectQuery';
import withOnError from 'src/common/withOnError';
import withTranslate from 'src/common/withTranslate';
import {compose} from 'src/helpers/utility';
import {GET_APP_LOCALE} from 'src/queries';
import MainPage from './MainPage';

const mapStateToProps = () => ({
  text: 'pages.main',
});

export default compose(
  withTranslate,
  connectQuery(GET_APP_LOCALE, mapStateToProps),
  withOnError,
)(MainPage);
