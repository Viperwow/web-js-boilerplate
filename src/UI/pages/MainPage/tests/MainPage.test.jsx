// Vendors
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
// UI
import MainPage from '..';

describe('MainPage component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<MainPage />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
