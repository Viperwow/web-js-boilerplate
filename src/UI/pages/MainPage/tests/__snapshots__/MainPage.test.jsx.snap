// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component should render correctly With default props 1`] = `
<div
  className="page"
>
  <ul>
    <li>
      <Link
        replace={true}
        to="/2"
      >
        2
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/3"
      >
        3
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/4"
      >
        4
      </Link>
    </li>
  </ul>
</div>
`;

exports[`MainPage component should render correctly With text prop 1`] = `
<div
  className="page"
>
  text
  <ul>
    <li>
      <Link
        replace={true}
        to="/2"
      >
        2
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/3"
      >
        3
      </Link>
    </li>
    <li>
      <Link
        replace={true}
        to="/4"
      >
        4
      </Link>
    </li>
  </ul>
</div>
`;
